packages/@aws-cdk/aws-appmesh/lib/virtual-node-listener.ts

```diff
@@ -0,0 +1,348 @@
+import * as cdk from '@aws-cdk/core';
+import { CfnVirtualNode } from './appmesh.generated';
+import { validateHealthChecks } from './private/utils';
+import { HealthCheck, HttpTimeout, TcpTimeout, Protocol, GrpcTimeout } from './shared-interfaces';
+
+/**
+ * Properties for a VirtualNode listener
+ */
+export interface VirtualNodeListenerConfig {
+  /**
+   * Single listener config for a VirtualNode
+   */
+  readonly listener: CfnVirtualNode.ListenerProperty,
+}
+
+/**
+ *  Defines listener for a VirtualNode
+ */
+export abstract class VirtualNodeListener {
+  /**
+   * Returns an HTTP Listener for a VirtualNode
+   */
+  public static httpNodeListener(props: HttpNodeListenerProps = {}): VirtualNodeListener {
```

Comment:
Same comment here as in https://github.com/aws/aws-cdk/pull/11277#discussion_r517000567. Should we call these `http` / `http2` / `grpc` / etc.?